socketcan_bridge/src/socketcan_to_topic.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2016, Ivor Wanders
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright notice,
+ *     this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the copyright holder nor the names of its
+ *     contributors may be used to endorse or promote products derived from
+ *     this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <socketcan_bridge/socketcan_to_topic.h>
+#include <socketcan_interface/string.h>
+#include <can_msgs/Frame.h>
+#include <string>
+
+namespace socketcan_bridge
+{
+  SocketCANToTopic::SocketCANToTopic(ros::NodeHandle* nh, ros::NodeHandle* nh_param,
+      boost::shared_ptr<can::DriverInterface> driver)
+    {
+      can_topic_ = nh->advertise<can_msgs::Frame>("received_messages", 10);
+      driver_ = driver;
+    };
+
+  int SocketCANToTopic::setup()
+    {
+      // register handler for frames and state changes.
+      frame_listener_ = driver_->createMsgListener(
+              can::CommInterface::FrameDelegate(this, &SocketCANToTopic::frameCallback));
+
+      state_listener_ = driver_->createStateListener(
+              can::StateInterface::StateDelegate(this, &SocketCANToTopic::stateCallback));
+    };
```

Comment:
return is missing.
What should be return? I would stick to bool (false means error) or change it to void.